### PR TITLE
Functional tests - Add test 'Update number of products displayed per page'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add functional test 'Update number of products displayed per page'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/13_shopParameters/03_productSettings/04_pagination/01_updateNumberOfProductsPerPage.js" URL_FO=shopUrl/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18317)

### CODE REVIEW

File: tests/puppeteer/pages/FO/category.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+module.exports = class Category extends FOBasePage {
+  constructor(page) {
+    super(page);
+
+    // Selectors
+    this.bodySelector = 'body#category';
+    this.mainSection = 'section#main';
+    this.productsSection = 'section#products';
+    this.productListDiv = 'div#js-product-list';
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/FO/category.js]:
```suggestion
    this.bodySelector = '#category';
    this.mainSection = 'section#main';
    this.productsSection = 'section#products';
    this.productListDiv = '#js-product-list';
```

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 2]
```diff
@@ -29,6 +29,10 @@ module.exports = class productSettings extends BOBasePage {
     // Products stock form
     this.productsStockForm = '#configuration_fieldset_stock';
     this.allowOrderingOosLabel = `${this.productsStockForm} label[for='form_stock_allow_ordering_oos_%TOGGLE']`;
+    // Pagination form
+    this.paginationFormBlock = '#configuration_fieldset_order_by_pagination';
+    this.productsPerPageInput = '#form_pagination_products_per_page';
+    this.savePaaginationFormButton = `${this.paginationFormBlock} .card-footer button`;
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]:
```suggestion
    this.savePaginationFormButton = `${this.paginationFormBlock} .card-footer button`;
```

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 3]
```diff
@@ -166,4 +170,15 @@ module.exports = class productSettings extends BOBasePage {
...
   }
+
+  /**
+   * Set number of products displayed per page
+   * @param numberOfProducts
+   * @return {Promise<string>}
+   */
+  async setProductsDisplayedPerPage(numberOfProducts) {
+    await this.setValue(this.productsPerPageInput, numberOfProducts.toString());
+    await this.clickAndWaitForNavigation(this.savePaaginationFormButton);
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(this.savePaginationFormButton);
```

File: tests/puppeteer/pages/FO/category.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    this.productItemListDiv = `${this.productListDiv} .products div[itemprop='itemListElement']`;
+  }
+
+  /* Methods */
+  /**
+   *
+   * @return {Promise<boolean>}
+   */
```
Comment (SimonGrn) [inline tests/puppeteer/pages/FO/category.js:20]: Maybe describe what these methods do ?

**Review by SimonGrn — COMMENTED** [review]
Thanks @boubkerbribri

File: tests/puppeteer/pages/FO/category.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/category.js]: done thanks

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]: done thanks

File: tests/puppeteer/pages/FO/category.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/category.js:20]: done thanks

**Review by SimonGrn — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**